packages/cursorless-engine/src/processTargets/modifiers/scopeHandlers/shouldYieldScope.ts

```diff
@@ -55,50 +64,34 @@ export function shouldYieldScope(
       break;
   }
 
-  // Don't yield scopes that end before the iteration is supposed to start
-  if (
-    direction === "forward"
-      ? domain.end.isBefore(position)
-      : domain.start.isAfter(position)
-  ) {
-    return false;
-  }
-
-  // Don't return non-empty scopes that end where the iteration is supposed to
-  // start
-  if (
-    !allowAdjacentScopes &&
-    !domain.isEmpty &&
-    (direction === "forward"
-      ? domain.end.isEqual(position)
-      : domain.start.isEqual(position))
-  ) {
-    return false;
-  }
+  return partiallyContains(
+    new Range(position, distalPosition),
+    domain,
+    allowAdjacentScopes,
+  );
```

Comment:
switching `distalPosition` to end of doc dramatically simplifies this code, and fixes several bugs along the way